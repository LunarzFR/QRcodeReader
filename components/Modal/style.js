import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  avatar: {
    borderWidth: 2,
    borderColor: '#20B4BA'
  },
  title: {
    fontWeight: 'bold',
    textAlign: 'center'
  },
  description: {
    fontSize: 20,
    textAlign: 'center',
    marginTop: '5%'
  }
})

export default styles;