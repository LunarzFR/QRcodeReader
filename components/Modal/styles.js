import { StyleSheet } from 'react-native';
import constants from '../../constants';

const styles = StyleSheet.create({
  avatar: {
    borderWidth: 2,
    alignSelf: 'center'
  },
  title: {
    fontWeight: 'bold',
    textAlign: 'center',
    marginTop: '5%'
  },
  description: {
    fontSize: 20,
    textAlign: 'center',
    marginTop: '5%'
  },
  button: {
    marginTop: '5%',
    width: '100%'
  }
});

export default styles;