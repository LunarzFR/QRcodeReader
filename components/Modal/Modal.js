import React from "react";
import { Overlay, Button } from "react-native-elements";
import Promotion from "./Promotion/Promotion";
import Error from "./Error/Error";
import { View } from "react-native";
import Icon from "react-native-vector-icons/FontAwesome";
import styles from "./styles";

const Modal = ({ modal, toggle, color }) => {
  return (
    <Overlay
      isVisible={modal.isVisible}
      height="50%"
      onBackdropPress={() => toggle()}
      height="auto"
    >
      <View style={{ alignItems: "center" }}>
        <View testID="promotion">
          {modal.type === "error" && <Error modal={modal} toggle={toggle}/>}
          {modal.type === "promotion" && (
              <Promotion  modal={modal} toggle={toggle} color={color}/>
          )}
        </View>
        <Button
          icon={<Icon name="undo" size={15} color="white"/>}
          title=" Go Back"
          buttonStyle={{ backgroundColor: color }}
          containerStyle={styles.button}
          onPress={() => toggle()}
        />
      </View>
    </Overlay>
  );
};

export default Modal;
