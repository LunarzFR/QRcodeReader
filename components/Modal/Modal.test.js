import React from "react";
import { fireEvent, render } from "react-native-testing-library";
import Modal from "./Modal";

describe("Testing Loading Spinner", () => {
  test("Show modal with different content", async () => {
    const { getByTestId, getByText, baseElement } = render(
      <Modal
        {...{
          toggle: () => {
            return;
          },
          modal: {
            type: "promotion",
            isVisible: false,
            data:
              {
                product: {
                  id: 1,
                  libelle: "SkateBoard",
                  url: "https://www.gifi.fr/media/catalog/product/cache/1/image/1000x/9df78eab33525d08d6e5fb8d27136e95/5/4/549274.jpg",
                  description: "petit skateboard des familles"
                }
              }
          }

        }}
      />
    );

    const button = getByText(" Go Back");
    fireEvent.press(button);

    expect(getByTestId("promotion").props.children.length).toBe(2);
    expect(baseElement).toMatchSnapshot();
  });
});
