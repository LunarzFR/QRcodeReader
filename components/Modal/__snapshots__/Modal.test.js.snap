// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Loading Spinner Show modal with different content 1`] = `undefined`;

exports[`examples of some things 1`] = `undefined`;
